### PR TITLE
Manage parameterized functional messages 

### PR DESCRIPTION
**Pull request by ayolab**
Signed-off-by: Ayoub LABIDI <ayoub.labidi@protonmail.com>

### CODE REVIEW

File: src/main/java/org/gridsuite/modification/server/ModificationType.java [hunk 1]
```diff
@@ -26,5 +26,9 @@ public enum ModificationType {
     LINE_SPLIT_WITH_VOLTAGE_LEVEL,
     LINE_ATTACH_TO_VOLTAGE_LEVEL,
     LINES_ATTACH_TO_SPLIT_LINES,
-    BRANCH_STATUS
+    BRANCH_STATUS,
+    LOCKOUT_LINE_CREATION,
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/modification/server/ModificationType.java:30]: Remove it
This enum represents only all modification types in DB

File: src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java [hunk 2]
```diff
@@ -90,6 +90,7 @@ public class NetworkModificationService {
     private static final String DELIMITER = "/";
 
     private static final String NETWORK_MODIFICATION_TYPE_REPORT = "NetworkModification";
+    private static final String LINE_ID = "lineId";
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java:93]: ?

File: src/test/java/org/gridsuite/modification/server/ModificationControllerTest.java [hunk 3]
```diff
@@ -582,10 +582,10 @@ public void testCreateLoadInNodeBreaker() throws Exception {
         mockMvc.perform(post(uriString, NOT_FOUND_NETWORK_ID).content(loadCreationInfosJson).contentType(MediaType.APPLICATION_JSON))
             .andExpectAll(status().isNotFound(), content().string(new NetworkModificationException(NETWORK_NOT_FOUND, NOT_FOUND_NETWORK_ID.toString()).getMessage())).andReturn();
 
-        loadCreationInfos.setEquipmentId(null);
+        loadCreationInfos.setEquipmentId("");
```
Comment (SlimaneAmar) [inline src/test/java/org/gridsuite/modification/server/ModificationControllerTest.java:585]: Add a test with null

File: src/main/java/org/gridsuite/modification/server/ModificationType.java (hunk 1 above)
Comment (SlimaneAmar) [reply src/main/java/org/gridsuite/modification/server/ModificationType.java:30]: Are you sure you have treated all types ?

**Review by SlimaneAmar — CHANGES_REQUESTED** [review]

File: src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java (hunk 2 above)
Comment (ayolab) [reply src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java:93]: I added this for the code smell : the "lineId" is used many times in createSubReporter calls for the different line creations

**Comment by etiennehomer** [comment]
Tests OK
After code smell fixes, it's good for me

File: src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java [hunk 4]
```diff
@@ -213,8 +214,9 @@ private List<ModificationInfos> execCreateLockoutLine(NetworkStoreListener liste
         Network network = listener.getNetwork();
         String rootReporterId = reporterId + "@" + NETWORK_MODIFICATION_TYPE_REPORT;
         ReporterModel reporter = new ReporterModel(rootReporterId, rootReporterId);
-        String subReportId = "Lockout line " + lineId;
-        Reporter subReporter = reporter.createSubReporter(subReportId, subReportId);
+        StringBuilder taskKey = new StringBuilder();
+        taskKey.append(ModificationType.BRANCH_STATUS.name()).append("_").append(ActionType.LOCKOUT.name());
+        Reporter subReporter = reporter.createSubReporter(taskKey.toString(), "Lockout line ${lineId}", LINE_ID_PARAMETER, lineId);
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java:219]: I prefer a concise writing for easy reading

  `              Reporter subReporter = reporter.createSubReporter(ModificationType.BRANCH_STATUS + "_" + ActionType.LOCKOUT, "Lockout line ${lineId}", LINE_ID_PARAMETER, lineId);`

File: src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java [hunk 5]
```diff
@@ -245,8 +247,9 @@ private List<ModificationInfos> execCreateTripLine(NetworkStoreListener listener
         Network network = listener.getNetwork();
         String rootReporterId = reporterId + "@" + NETWORK_MODIFICATION_TYPE_REPORT;
         ReporterModel reporter = new ReporterModel(rootReporterId, rootReporterId);
-        String subReportId = "Trip line " + lineId;
-        Reporter subReporter = reporter.createSubReporter(subReportId, subReportId);
+        StringBuilder taskKey = new StringBuilder();
+        taskKey.append(ModificationType.BRANCH_STATUS.name()).append("_").append(ActionType.TRIP.name());
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java:251]: Same

File: src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java [hunk 6]
```diff
@@ -286,8 +289,10 @@ private List<ModificationInfos> execCreateEnergiseLineEnd(NetworkStoreListener l
         Network network = listener.getNetwork();
         String rootReporterId = reporterId + "@" + NETWORK_MODIFICATION_TYPE_REPORT;
         ReporterModel reporter = new ReporterModel(rootReporterId, rootReporterId);
-        String subReportId = "Energise line " + lineId;
-        Reporter subReporter = reporter.createSubReporter(subReportId, subReportId);
+        String branchSideKey = side == Branch.Side.ONE ? ActionType.ENERGISE_END_TWO.name() : ActionType.ENERGISE_END_ONE.name();
+        StringBuilder taskKey = new StringBuilder();
+        taskKey.append(ModificationType.BRANCH_STATUS.name()).append("_").append(branchSideKey);
+        Reporter subReporter = reporter.createSubReporter(taskKey.toString(), "Energise line ${lineId}", LINE_ID_PARAMETER, lineId);
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java:295]: Same

**Review by SlimaneAmar — APPROVED** [review]

**Review by etiennehomer — APPROVED** [review]
